libs/s25main/desktops/dskOptions.cpp

```diff
@@ -663,3 +688,91 @@ void dskOptions::loadVideoModes()
     // Sort by aspect ratio
     std::sort(video_modes.begin(), video_modes.end(), cmpVideoModes);
 }
+
+void dskOptions::Msg_ScreenResize(const ScreenResizeEvent& sr)
+{
+    Desktop::Msg_ScreenResize(sr);
+    updateGuiScale();
+}
+
+bool dskOptions::Msg_WheelUp(const MouseCoords& mc)
+{
+    if(!scrollGuiScale(true))
+        return Desktop::Msg_WheelUp(mc);
+    return true;
+}
+
+bool dskOptions::Msg_WheelDown(const MouseCoords& mc)
+{
+    if(!scrollGuiScale(false))
+        return Desktop::Msg_WheelDown(mc);
+    return true;
+}
+
+void dskOptions::updateGuiScale(ctrlComboBox* combo)
+{
+    // generate GUI scale percentages in 10% increments
+    constexpr auto stepSize = 10u;
+    const auto roundGuiScale = [=](unsigned percent) { return (percent + stepSize / 2) / stepSize * stepSize; };
+
+    unsigned minPercent, maxPercent, recommendedPercent;
+    VIDEODRIVER.getGuiScaleRange(&minPercent, &maxPercent, &recommendedPercent);
+    const auto recommendedPercentRounded = roundGuiScale(recommendedPercent);
+
+    if(!combo)
+    {
+        auto* group = GetCtrl<ctrlGroup>(ID_grpGraphics);
+        combo = group->GetCtrl<ctrlComboBox>(ID_cbGuiScale);
+    }
+
+    guiScales_.clear();
+    combo->DeleteAllItems();
+
+    combo->AddString((boost::format(_("Auto (%u%%)")) % recommendedPercent).str());
+    if(SETTINGS.video.guiScale == 0)
+        combo->SetSelection(0);
+
+    for(unsigned percent = roundGuiScale(minPercent); percent <= maxPercent; percent += stepSize)
+    {
+        if(percent == recommendedPercentRounded)
+            recommendedGuiScaleIndex_ = guiScales_.size();
+        guiScales_.push_back(percent);
+
+        combo->AddString(helpers::toString(percent) + "%");
+        if(percent == SETTINGS.video.guiScale)
+            combo->SetSelection(combo->GetNumItems() - 1);
```

Comment:
Might be easier to read if you swap the lines a bit:
```suggestion
        if(percent == SETTINGS.video.guiScale)
            combo->SetSelection(combo->GetNumItems());

        combo->AddString(helpers::toString(percent) + "%");
        guiScales_.push_back(percent);
```

Reply:
Yeah, I assume `SetSelection()` won't allow me to go past the number of items.